packages/astro/src/assets/fonts/utils.ts

```diff
@@ -83,24 +92,10 @@ export type CacheHandler = ReturnType<typeof createCache>;
  * - `collect` will save the association of the original url and the new hash for later use
  * - the returned url will be `/_astro/fonts/<hash>.woff2`
  */
-export function createURLProxy({
-	hashString,
-	collect,
-}: {
-	hashString: (value: string) => string;
-	collect: (data: {
-		hash: string;
-		type: FontType;
-		value: string;
-	}) => string;
-}) {
-	return function proxyURL(value: string): string {
-		const type = extractFontType(value);
-		const hash = `${hashString(value)}.${type}`;
-		const url = collect({ hash, type, value });
-		// Now that we collected the original url, we return our proxy so the consumer can override it
-		return url;
-	};
+export function proxyURL(value: string, { hashString, collect }: ProxyURLOptions): string {
```

Comment:
Also, why do we pass a string and an object? Wouldn't be easier to just pass everything inside the object? Sorry, they might be silly questions, but it seems that the complexity of the code increasing without reasons